src/rvsdg/to_cfg.rs

```diff
@@ -0,0 +1,523 @@
+                    .collect::<Vec<_>>();
+
+                // evaluate pred in this block as well
+                // TODO we are assuming pred is an int here, is that actually true?
+                let pred = self.operand_to_bril(*pred, current_args, ctx);
+                let pred_bool = self.cast_bool(&pred);
+                let prev_block = self.finish_block();
+
+                let mut branch_blocks = vec![];
+                let mut shared_vars = None;
+                // for each set of outputs in outputs, make a new block for them
+                for outputs in outputs {
+                    // evaluate this branch
+                    let resulting_outputs = outputs
+                        .iter()
+                        .map(|operand| self.operand_to_bril(*operand, &input_vars, &Some(id)))
+                        .collect::<Vec<_>>();
+                    if shared_vars.is_none() {
+                        shared_vars = Some(self.fresh_variables_for(&resulting_outputs));
+                    }
+                    // assign to the shared vars
+                    self.assign_to_vars(&resulting_outputs, shared_vars.as_ref().unwrap());
+
+                    branch_blocks.push(self.finish_block());
+                }
+
+                // we need to conditionally jump to each of the branch blocks
+                // based on the predicate
+                // TODO right now we
+                // just handle the case where we branch to two things
+                assert!(outputs.len() == 2);
+                assert!(branch_blocks.len() == 2);
+
+                // add a conditional jump from the previous block to the branch blocks
+                self.graph.add_edge(
+                    prev_block,
+                    branch_blocks[0],
+                    Branch {
+                        op: BranchOp::Cond {
+                            arg: Identifier::Name(pred_bool.clone()),
+                            val: false.into(),
+                        },
+                        pos: None,
+                    },
+                );
+                self.graph.add_edge(
+                    prev_block,
+                    branch_blocks[1],
+                    Branch {
+                        op: BranchOp::Cond {
+                            arg: Identifier::Name(pred_bool),
+                            val: true.into(),
+                        },
+                        pos: None,
+                    },
+                );
+
+                // now we have all the branches, make incomplete jumps for each of them
+                for branch_block in branch_blocks {
+                    self.incomplete_branches
+                        .push(IncompleteBranch::Direct { from: branch_block });
+                }
+
+                shared_vars.unwrap()
+            }
+            RvsdgBody::Theta {
+                pred: pred_operand,
+                inputs,
+                outputs,
+            } => {
```

Comment:
I find this hard to understand. Some comments (or even ASCII art) for how the CFG layout for Theta node will be very helpful!

Reply:
Done!